src/pymor/reductors/neural_network.py

```diff
@@ -175,33 +198,46 @@ def reduce(self, hidden_layers='[(N+P)*3, (N+P)*3]', activation_function=torch.t
 
         torch.manual_seed(get_seed_seq().spawn(1)[0].generate_state(1).item())
 
-        # build a reduced basis using POD and compute training data
-        if not hasattr(self, 'training_data'):
-            self.compute_training_data()
+        # compute training snapshots
+        if self.training_snapshots is None:
+            self.compute_training_snapshots()
 
-        layer_sizes = self._compute_layer_sizes(hidden_layers)
+        # build a reduced basis using POD
+        if self.reduced_basis is None:
+            self.compute_reduced_basis()
+
+        # compute training data
+        if self.training_data is None:
+            self.compute_training_data()
+        assert self.training_data is not None
+        assert len(self.training_data) == len(self.training_set) * self.nt
+
+        if self.validation_set is None:
+            number_validation_snapshots = int(len(self.training_data) * self.validation_ratio)
+            # randomly shuffle training data before splitting into two sets
+            get_rng().shuffle(self.training_data)
+            # split training snapshots into validation and training snapshots
+            self.validation_data = self.training_data[0:number_validation_snapshots]
+            self.validation_set = [data[0] for data in self.validation_data]
+            self.training_data = self.training_data[number_validation_snapshots:]
+
```

Comment:
should we also update the ```training_set```, ```training_snapshots``` and ```validation_snapshot``` in here just to be safe for potential bugs in the future? 
Right now it only uses the ```validataion_data``` from this point onwards.